app/actions/local/group.ts

```diff
@@ -23,16 +18,12 @@ export const searchGroupsByName = async (serverUrl: string, name: string): Promi
     } catch (e) {
         // eslint-disable-next-line no-console
         console.log('searchGroupsByName - ERROR', e);
-        return queryGroupsByName(operator.database, name).fetch();
+        const {database} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
```

Comment:
This still may throw. Maybe put it where the operator used to be? Same comment on several other places on the file.

Reply:
Yup I was wondering about this - do we want `try{ .. } catch() { ... try/catch{} }`? Nested try/catch inside catch feels like an anti-pattern.

Reply:
Yeah... nested try catch sounds wrong. We can get the database at the beginning of the function, the same way we do sometimes with the client. i.e.
```
function x () {
    let database;
    try {
        database = DatabaseManager.getServerDatabaseAndOperator(serverUrl).database;
    } catch () {
        return {error: 'cannot find database'}
    }

    ...
}
```
In this case where we only need the database on the catch, instead of returning the error, we can do nothing on the catch, and just check at that point if database is defined or not, and handle it accordingly.